api/src/utils/apply-query.ts

```diff
@@ -765,14 +765,32 @@ export async function applySearch(
 				this.orWhereRaw(`LOWER(??) LIKE ?`, [`${collection}.${name}`, `%${searchQuery.toLowerCase()}%`]);
 			} else if (['bigInteger', 'integer', 'decimal', 'float'].includes(field.type)) {
 				const number = Number(searchQuery);
-				if (!isNaN(number)) this.orWhere({ [`${collection}.${name}`]: number });
+
+				// only cast finite base10 numeric values
+				if (validateNumber(searchQuery, number)) {
+					this.orWhere({ [`${collection}.${name}`]: number });
+				}
 			} else if (field.type === 'uuid' && validate(searchQuery)) {
 				this.orWhere({ [`${collection}.${name}`]: searchQuery });
 			}
 		});
 	});
 }
 
+function validateNumber(value: string, parsed: number) {
+	if (isNaN(parsed) || !Number.isFinite(parsed)) return false;
+
+	// compare without the decimal point
+	if (value.includes('.')) {
+		value = value.replace('.', '');
+		parsed = Number(value);
+	}
```

Comment:
Am I correctly reading this as:

```
filter: {
  myIntegerField: {
    _eq: 12.5
  }
}
```

will now compare to the value `125`?

Reply:
Only when using `?search=12.5` then it is compared like `String(Number('125')) === '125'` to catch the specific case of decimals with `0` like `?search=42.00` which would be parsed and compared like `'42' === '42.00'` otherwise. This value is only used for the check inside this function and does not apply to filters or other query elements.

Reply:
Perhaps i should have just plopped this regex `/\d+\.?\d*/` in there for the string validation 🤔

Reply:
I think it's safer to error out if the `.` is encountered, rather than trying to be "smart" about it 👍🏻

Reply:
Hmm I agree that it seems a little obscure (maybe mention the `42.00` example in the comment) , but I think it should be safe since these lines will only be called when the value is a number.

Reply:
Not saying it isn't safe, just saying that it's unexpected. I'd rather have the one rule that says "You can't use floats to search for integers", than have the compound setup of "You can use floats, but they'll be converted to integers by dropping the decimal point, which will work for `42.00`, but causes unexpected behavior for `12.5`" 🙂

Reply:
Ah gotcha! I agree that it invokes unexpected behavior 👍 Although it is the other way around: Right now `"42.00"` would be unexpectedly converted to `42`, but `"12.5"` would be `12.5` as expected.

So we can just drop this comparison and we're good to go, right?

Reply:
Yeah I think so 👍🏻

Reply:
Do you want to drop float searching? i would expect if i searched `12.5` that i would find any `decimal` or `float` field containing `12.5` 🤔 This validation applies to numeric types not just `integer`:
```js
} else if (['bigInteger', 'integer', 'decimal', 'float'].includes(field.type)) {
```

Reply:
No, just to not handle the cases when the search value contains only zeros after the decimal point, like `42.00`.

AFAICS should be possible by the following proposal:

> So we can just drop this comparison (line 784-787) and we're good to go, right?

Reply:
I see, fair it is a bit of extra complexity for this one edgecase